changelog.d/20240819_162411_fnareoh_scrt_4759_optimize_files_to_scan_on_merge_commits.md

```diff
@@ -0,0 +1,5 @@
+### Added
+
+- The command `ggshield secret scan pre-commit` has a new flag `--merge-skip-unchanged`. It is off by default, if activated,
```

Comment:
I think feared `skip-unchanged-merge` would make it sound like we skip entirely the merge commit when we only skip specific unchanged files ?  
For me as a feature that reduce the amount of what we scan, I would go for safety first and keep it as OFF by default. Plus it would mean keeping the current behavior as default which makes sense to me. 

But I don't have a strong opinion either and it has not been clearly discussed with Mathieu.

Reply:
Alright, I understand. This won't be part of the 1.31.0 scheduled for tomorrow anyway, so we can still discuss this with Mathieu before the next release :)

Reply:
The name is not optimal, but I don't have anything better to propose yet. Need to think this through.

As for it being by default, I think it's better to make it opt-in: this flag is only safe if you are confident that all branches you are about to merge have already been scanned. I don't think this is true in all situations.

Reply:
After discussion with @mathieubellon `--skip-unchanged-merge-files` is good, long but clear and for now we keep it as an option deactivated by default but in the future we may change it to activate it by default (as well as other flags) to improve default performances.